saturn-datepicker/src/datepicker/calendar-body.ts

```diff
@@ -120,6 +128,14 @@ export class SatCalendarBody implements OnChanges {
     }
   }
 
+  _mouseOverCell(cell: SatCalendarCell): void {
+    if (this.beginSelected && this.rangeMode) {
+      this._cellOver = cell.value;
+    } else {
+      this._cellOver = null;
+    }
```

Comment:
Just:
```
    this._cellOver = cell.value;
```
I know, it affects your logic below, but I think it will make your code more understandable